File: app/src/main/java/closer/vlllage/com/closer/store/models/Group.java
package closer.vlllage.com.closer.store.models;

import io.objectbox.annotation.Entity;

@Entity
public class Group extends BaseObject {
    private String name;
    private String about;
    private boolean isPublic;
    private String eventId;
    private Double latitude;
    private Double longitude;

    public String getName() {
        return name;
    }

    public Group setName(String name) {
        this.name = name;
        return this;
    }

    public String getAbout() {
        return about;
    }

    public Group setAbout(String about) {
        this.about = about;
        return this;
    }

    public boolean isPublic() {
        return isPublic;
    }

    public Group setPublic(boolean aPublic) {
        isPublic = aPublic;
        return this;
    }

    public Double getLatitude() {
        return latitude;
    }

    public Group setLatitude(Double latitude) {
        this.latitude = latitude;
        return this;
    }

    public Double getLongitude() {
        return longitude;
    }

    public Group setLongitude(Double longitude) {
        this.longitude = longitude;
        return this;
    }

    public String getEventId() {
        return eventId;
    }

    public Group setEventId(String eventId) {
        this.eventId = eventId;
        return this;
    }

    public boolean hasEvent() {
        return eventId != null;
    }
}

File: app/src/main/java/closer/vlllage/com/closer/store/models/Group.kt
package closer.vlllage.com.closer.store.models

import io.objectbox.annotation.Entity

@Entity
class Group : BaseObject() {
    var name: String? = null
    var about: String? = null
    var ratingAverage: Double? = null
    var ratingCount: Int? = null
    var isPublic: Boolean = false
    var direct: Boolean = false
    var hub: Boolean = false
    var physical: Boolean = false
    var eventId: String? = null
    var phoneId: String? = null
    var ofId: String? = null
    var ofKind: String? = null
    var groupMessageId: String? = null
    var latitude: Double? = null
    var longitude: Double? = null
    var photo: String? = null

    fun hasEvent() = eventId != null
    fun hasPhone() = phoneId != null
    fun hasGroupMessage() = groupMessageId != null
}
